src/utils/identity/parsePem.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  Ed25519KeyIdentity
+} from '@dfinity/identity';
+import Secp256k1KeyIdentity from './secpk256k1/identity';
+import { Types } from '../account/constants';
+
+export const getIdentityFromPem = (pem) => {
+  var raw;
+
+  pem = pem
+    .replace(/(-{5}.*-{5})/g, "")
+    .replace("\n", "")
+    // Sepk256k1 keys
+    .replace("BgUrgQQACg==", "")
+    .trim();
+
+  raw = Buffer.from(pem, "base64")
+    .toString("hex")
+    // Sepk256k1 keys
+    .replace("30740201010420", "")
+    .replace("a00706052b8104000aa144034200", "")
+    // ED25519 keys
+    .replace("3053020101300506032b657004220420", "")
+    .replace("a123032100", "");
+  var key;
+  var identity;
+  var type;
+
+  try {
+    key = new Uint8Array(Buffer.from(raw, "hex"));
+    identity = Ed25519KeyIdentity.fromSecretKey(key);
+    type = Types.pem25519;
+  } catch {
+    try {
+      key = new Uint8Array(Buffer.from(raw.substring(0, 64), "hex"))
+      identity = Secp256k1KeyIdentity.fromSecretKey(key);
+      type = Types.pem256k1;
+    } catch (e) {
+      console.log(e, "(e) Invalid key");
+      process.exit(1);
```

Comment:
Check this. I think it should throw some error and catch it where u call it or let it go up